File: space-game/Core.Tests/Extensions/GuidExtensionsTests.vb
Imports Core.Extensions
Imports NUnit.Framework

Namespace Extensions
    <TestFixture>
    Public Class GuidExtensionsTests

        <Test()>
        Public Sub IsEmpty_WhenEmpty_ReturnsTrue()
            Dim test As Guid

            Dim result = test.IsEmpty()

            Assert.AreEqual(True, result)
        End Sub

        <Test()>
        Public Sub IsEmpty_WhenNotEmpty_ReturnsFalse()
            Dim test = Guid.NewGuid()

            Dim result = test.IsEmpty()

            Assert.AreEqual(False, result)
        End Sub

    End Class
End Namespace

File: space-game/Core.Tests/Extensions/GuidExtensionsTests.cs
using Core.Extensions;
using NUnit.Framework;
using System;

namespace Core.Tests.Extensions
{
    [TestFixture]
    public class GuidExtensionsTests
    {
        [Test]
        public void IsEmpty_WhenEmpty_ReturnsTrue()
        {
            Guid test = default(Guid);

            bool result = test.IsEmpty();

            Assert.AreEqual(true, result);
        }

        [Test]
        public void IsEmpty_WhenNotEmpty_ReturnsFalse()
        {
            Guid test = Guid.NewGuid();

            bool result = test.IsEmpty();

            Assert.AreEqual(false, result);
        }
    }
}
